src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -525,40 +534,127 @@ class OpenShiftDSL implements Serializable {
      * When labels are used, the qualifier will be a map. In other cases, expect a String or null.
      */
     public OpenShiftResourceSelector selector(Object kind = null, Object qualifier=null) {
-        return new OpenShiftResourceSelector("selector", kind, qualifier);
+        OpenShiftResourceSelector selector = new OpenShiftResourceSelector("selector", kind, qualifier);
+        selector.projectList = new ArrayList<String>();
+        for (int i=0; i < selector.objectList.size(); i++) {
+            selector.projectList.add(project());
+        }
+        return selector;
     }
 
     private OpenShiftResourceSelector objectDefAction(String verb, Object obj, Object[] userArgs) {
+        dieIfWithout(currentContext, ContextId.WITH_CLUSTER, verb);
+        // if the obj is broken down to a parseable set of maps (typically as the result of an
+        // openshift.process call, we look to see if the namespace is specified for any of them
+        // or if they are not consistently the same namespace:
+        // no namespaces set: we make 1 invocation of the verb where we will use the current project stored in
+        // the context if it exists (traditional behavior)
+        // consistent namespace set: we make 1 invocation of the verb where we don't use the current project stored in
+        // the context and let oc and the obj json sort out where objects go
+        // inconsistent namespaces: we make an invocation of the verb for each object where will will not override the 
+        // project if it is set but use the current project if it is not set
+        Object namespace = currentContext.getProject(); // set default in case not list
+        boolean found = false;
+        boolean consistent = true;
+        ArrayList objList = null;
+        if (obj instanceof List) {
+            objList = new ArrayList((List)obj);
+            for (Object obj2 : objList) {
+                if (obj2 instanceof Map) {
+                    HashMap obj2Map = new HashMap((Map)obj2);
+                    HashMap obj2MapMetadata = obj2Map.get("metadata");
+                    if (obj2MapMetadata != null) {
+                        Object ns = obj2MapMetadata.get("namespace");
+                        // first time through the list, no 
+                        // consistency check
+                        if (!found) {
+                            found = true;
+                            namespace = ns;
+                        } else {
+                            if (namespace == null && ns == null)
+                                continue;
+                            if (namespace != null && namespace.equals(ns))
+                                continue;
+                            if (ns != null && ns.equals(namespace))
```

Comment:
this can only be true if the previous check was also true, so it's redundant, right?  (Unless someone screwed up the reflexive logic on an equals implementation)

Reply:
yep the previous 2 checks make this redundant ... I'll update